### PR TITLE
Evelinas todo/notes-app

### PR DESCRIPTION
**Pull request by evelinabe**
[Link to my deployed project](https://notes-todo.netlify.app/)

My reflections on how this week's project turned out:
I took advantage of this week's artistic freedom. I had a vision of the todo-app of my dreams and sort of made it come true. This means I pretty sailed away and did not so much follow the requirements. So I don't have a toggle for completed items, but I do have a multipage todo app where I can edit the tasks (YAY!). It was loads of fun to sail off!

### CODE REVIEW

File: code/src/components/AddTodoForm.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { useHistory } from "react-router-dom";
+import uniqid from "uniqid";
```
Comment (emma-tech) [inline code/src/components/AddTodoForm.js]: Really cool that you tried this out this with uniqid that Maks showed us in his code from last week, gonna ask you more about how it works some time 🤩

File: code/src/components/AddTodoForm.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  const breakpoint = 767;
+  const dispatch = useDispatch();
+  const categories = useSelector((store) => store.todos.categories);
+  const history = useHistory();
+
+  useEffect(() => {
+    window.addEventListener("resize", () => setWidth(window.innerWidth));
+  }, []);
+
+  console.log(categories[1].subcategories);
```
Comment (emma-tech) [inline code/src/components/AddTodoForm.js]: Don't forget to remove console.logs! 😊

File: code/src/components/AddTodoForm.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+  const onCategoryChange = (categoryValue) => {
+    userCategory.includes(categoryValue)
+      ? setUserCategory(userCategory.filter((item) => item !== categoryValue))
+      : setUserCategory([...userCategory, categoryValue]);
+  };
+  console.log(userCategory);
```
Comment (emma-tech) [inline code/src/components/AddTodoForm.js]: Another one ;)

File: code/src/components/AddTodoForm.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      setTodoTitle("");
+      setTodoContent("");
+      setUserCategory([]);
+      history.push("/");
+    } else {
+      alert("Please write note first");
+    }
+  };
+
+  return (
```
Comment (emma-tech) [inline code/src/components/AddTodoForm.js]: Really nice and well structured code! Easy to read and understand 🥇

File: code/src/components/EditTodoForm.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+          />
+        </TodoInputGroupWrapper>
+
+        <TodoInputGroupWrapper>
+          <TodoContentLabel htmlFor="todoContent">p</TodoContentLabel>
+          <TodoTextarea
+            type="textarea"
+            id="todoContent"
+            value={todoContent}
+            aria-label="Write your task here"
```
Comment (emma-tech) [inline code/src/components/EditTodoForm.js]: Good that you use aria-labels 👏

File: code/src/components/EditTodoForm.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            rows="10"
+            onChange={(e) => setTodoContent(e.target.value)}
+          />
+        </TodoInputGroupWrapper>
+
+        <TodoCheckboxGroupWrapper>
```
Comment (emma-tech) [inline code/src/components/EditTodoForm.js]: Wooow, your project is so advanced! 🤩 really nice and well structured code, good job! 🌟

File: code/src/components/Item.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { useDispatch } from "react-redux";
+import { todos } from "reducers/todos";
+import { Link } from "react-router-dom";
+import DeleteOutlinedIcon from "@material-ui/icons/DeleteOutlined";
+
+import { TimeAgo } from "./TimeAgo";
+import {
+  ListItem,
```
Comment (emma-tech) [inline code/src/components/Item.js:8]: Really cool that you used this with lib for structuring and importing your styling, I'm gonna try to use this approach in next project  👏

File: code/src/lib/AddTodoIcon.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import styled from "styled-components";
+import { Link } from "react-router-dom";
+import PostAddIcon from "@material-ui/icons/PostAddRounded";
+
+import breakpoints from "./Breakpoints";
+
+const CustomLinkIconContainer = styled.div`
+position: absolute;
```
Comment (emma-tech) [inline code/src/lib/AddTodoIcon.js]: Missing some indents on these rows :)

File: code/src/lib/FormStyle.js [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+    display: flex;
+    align-items: center;
+  }
+  @media only screen and ${breakpoints.device.tablet} {
+    justify-content: center;
+  }
+`;
+
+export const TodoForm = styled.form`
+width: 100%;
```
Comment (emma-tech) [inline code/src/lib/FormStyle.js]: Just a little bit of indent missing here :)

File: code/src/lib/FormStyle.js [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+  background-color: #fafafa;
+  @media only screen and ${breakpoints.device.tablet} {
+    padding-top: 40px:
+  }
+  @media only screen and ${breakpoints.device.computer} {
+    width: 85%;
+  }
+`;
+
+export const TodoInputGroupWrapper = styled.div`
```
Comment (emma-tech) [inline code/src/lib/FormStyle.js:30]: I just LOVE your styling 🤩 It's so beautiful and clean, really cool design 👏

File: code/src/reducers/todos.js [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+      title: "second",
+      content: "",
+      category: ["#Free time"],
+      createdAt: 1604676513000,
+    },
+    {
+      id: "3a",
+      title: "third",
+      content: "Bake cookies with kids",
+      category: ["#Shop"],
```
Comment (emma-tech) [inline code/src/reducers/todos.js]: Really nice that you've implemented categories 👏 (wouldn't it be cool if you can see them on the tasks visible on the start page as well? :D )

File: code/src/reducers/todos.js [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+        content,
+        category,
+        createdAt,
+      } = action.payload;
+      const currentTodo = store.items.find((item) => item.id === id);
+      if (currentTodo) {
+        currentTodo.title = title;
+        currentTodo.content = content;
+        currentTodo.category = category;
+        currentTodo.createdAt = createdAt;
```
Comment (emma-tech) [inline code/src/reducers/todos.js]: So cool that you implemented time created as well 👏

File: code/src/reducers/todos.js [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+      content: "Bake cookies with kids",
+      category: ["#Shop"],
+      createdAt: 1606318113000,
+    },
+  ],
+  addItem: false,
+  categories,
+};
+
+export const todos = createSlice({
```
Comment (emma-tech) [inline code/src/reducers/todos.js:65]: Really nice and well structured code, It's easy to follow! 🌟

File: code/src/lib/EditIcon.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import styled from "styled-components";
+import EditOutlinedIcon from "@material-ui/icons/EditOutlined";
```
Comment (emma-tech) [inline code/src/lib/EditIcon.js]: I love the icons! 🤩👏

File: code/src/lib/AddTodoIcon.js [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+  cursor: pointer;
+  color: white;   
+`;
+
+export const AddTodoIcon = () => {
+  return (
+    <CustomLinkIconContainer>
+      <Link to={`/todo/`}>
+        <CustomLinkIcon>
+          <PostAddIcon style={{ fontSize: 32 }} />
```
Comment (emma-tech) [inline code/src/lib/AddTodoIcon.js]: (Just one little thing I was thinking about: when adding a lot of tasks in your list the add todo icon "gets in the way"/ is sitting on top of of the tasks. Maybe consider moving it above the list? :D )

File: code/src/lib/FormStyle.js [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+  @media only screen and ${breakpoints.device.tablet} {
+    font-size: 22px;
+    margin: 10px 14px 0 30px;
+  }
+  @media only screen and ${breakpoints.device.computer} {
+    font-size: 16px;
+  }
+`;
+
+export const TodoTextarea = styled.textarea`
```
Comment (emma-tech) [inline code/src/lib/FormStyle.js:97]: Your naming of components, classes and reducers all trough the project is really good Evelina, I have no trouble figuring out what they do 👍  good job 💯

File: code/src/App.js [hunk 17]
```diff
@@ -1,9 +1,35 @@
...
+          </Route>
+          <Route exact path="/todo">
+            <AddTodoForm />
+          </Route>
+          <Route path="/todo/:id">
+            <EditTodoForm />
+          </Route>
+        </Switch>
+      </BrowserRouter>
+      {/* <AddTodoForm></AddTodoForm>
```
Comment (emma-tech) [inline code/src/App.js]: I think you can remove this out commented code (unless it isn't something you're planning on using later of course! 👍)

File: code/src/lib/ListHeader.js [hunk 18]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import styled from "styled-components";
+
+import logo from "./scribble-icon-21.jpg";
```
Comment (emma-tech) [inline code/src/lib/ListHeader.js:4]: Love the logo 🤩 (wouldn't it be really cool if you could add it as a favicon as well? 🤩)

**Review by emma-tech — COMMENTED** [review]
Evelina! I loved reviewing your project 🤩 Your todo app is so advanced and well styled, and I can really tell that you had so much fun with this project 🤩 (I loved that you "sailed away" a bit 😂👏) Really good job with the responsive styling 👍  
Also really cool that you made it into a multipage application and that you implemented so many features like the icons, categories, dates and the input for heading and text, it makes the app look really classy and advanced 👏 GOOD JOB 🤩

**Review by emma-tech — APPROVED** [review]
🌟 Your app is so cool Evelina! I can really tell you put a lot of time and energy into it, and that you had fun creating it 🤩 (you wrote in your pull request that you don't have a toggle for completed items, but this is something that I think you have no trouble adding later if you would want to), and there already are so much other cool and advanced stuff here, So I say it's APPROVED 🌟

File: code/src/components/AddTodoForm.js [hunk 19]
```diff
@@ -0,0 +1,133 @@
...
+                          <CheckboxInput
+                            type="checkbox"
+                            name={subcategory.name[index]}
+                            value={subcategory.name}
+                            checked={userCategory.includes(subcategory.name)}
+                            onChange={() => onCategoryChange(subcategory.name)}
+                          />
+                          {subcategory.name}
+                        </CheckboxLabel>
+                      </SubCheckboxWrapper>
```
Comment (HIPPIEKICK) [inline code/src/components/AddTodoForm.js]: Nice naming! Maybe style the categories and the subcategories different as one might not (I didn't 🙈)understand the levels in the existing design

File: code/src/components/EditTodoForm.js [hunk 20]
```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect } from "react";
```
Comment (HIPPIEKICK) [inline code/src/components/EditTodoForm.js]: The code in this file look very familiar 🤔 What can we do about that?

File: code/src/reducers/todos.js (hunk 12 above)
Comment (HIPPIEKICK) [reply code/src/reducers/todos.js]: Yes, this is very impressive!! I really like that you came up with this idea! In the requirements for this week - you forgot a thing though. When you have time - implement this as well:
_You should be able to mark an uncompleted task as complete (and change it back to uncomplete)_

**Comment by evelinabe** [comment]
@HIPPIEKICK Hi, have you had the time to look at my changes yet?

**Pull request closed without merging**